src/test/java/org/opensearch/knn/integ/MixedVectorDocumentIT.java

```diff
@@ -0,0 +1,594 @@
+        } catch (ResponseException e) {
+            fail("Search failed for index " + indexName + ": " + e.getMessage());
+            return null;
+        }
+    }
+
+    private String performCountAndGetBody(String indexName) throws IOException, ParseException {
+        try {
+            return EntityUtils.toString(performRequest("GET", "/" + indexName + "/_count").getEntity());
+        } catch (ResponseException e) {
+            fail("Count request failed for index " + indexName + ": " + e.getMessage());
+            return null;
+        }
+    }
+
+    private String performSearchCountAndGetBody(String indexName, XContentBuilder query) throws IOException, ParseException {
+        try {
+            Request request = new Request("POST", "/" + indexName + "/_count");
+            request.setJsonEntity(query.toString());
+            return EntityUtils.toString(client().performRequest(request).getEntity());
+        } catch (ResponseException e) {
+            fail("Search count request failed for index " + indexName + ": " + e.getMessage());
+            return null;
+        }
+    }
+
+    private Response performKNNSearch(String indexName, float[] queryVector, int k) throws IOException {
+        XContentBuilder query = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("query")
+            .startObject("knn")
+            .startObject(VECTOR_FIELD)
+            .field("vector", queryVector)
+            .field("k", k)
+            .endObject()
+            .endObject()
+            .endObject()
+            .endObject();
+        return performSearch(indexName, query);
+    }
+
+    private Response performSearch(String indexName, XContentBuilder query) throws IOException {
+        Request request = new Request("POST", "/" + indexName + "/_search");
+        request.setJsonEntity(query.toString());
+        return client().performRequest(request);
+    }
+
+    private Response performRequest(String method, String endpoint) throws IOException {
+        return client().performRequest(new Request(method, endpoint));
+    }
+
+    private void updateIndexSettings(String indexName, String settings) throws IOException {
+        Request request = new Request("PUT", "/" + indexName + "/_settings");
+        request.setJsonEntity("{" + settings + "}");
+        client().performRequest(request);
+    }
+
+    private float[] createRandomVector(int dimension) {
+        float[] vector = new float[dimension];
+        for (int i = 0; i < dimension; i++) {
+            vector[i] = (float) (Math.random() * 10.0);
+        }
+        return vector;
+    }
+
+    private boolean checkIndexExists(String indexName) throws IOException {
+        try {
+            return client().performRequest(new Request("HEAD", "/" + indexName)).getStatusLine().getStatusCode() == 200;
+        } catch (ResponseException e) {
+            return false;
+        }
+    }
```

Comment:
can we move this helper functions in the base class. I think some of the functions should already be there